File: learning/ojayer/Py_Demo_3/Service.py
#import libraries for abstraction
from abc import ABC, abstractmethod


class Service(ABC):
    #abstract methods for service classes
    @abstractmethod
    def authenticate(self, bucket_number: int):
        pass

    @abstractmethod
    def listFiles(self, max_results: int = None, query: str = None):
        pass

    @abstractmethod
    def check_storage(self):
        pass

File: learning/rihal/Mobile App Initial/syncly/app/src/main/java/com/example/syncly/Service.java
package com.example.syncly;
import java.util.Map;
import java.util.List;

public abstract class Service {

    public interface AuthCallback {
        void onAuthComplete(Object result);
        void onAuthFailed(String error);
    }

    public interface ListFilesCallback {
        void onFilesListed(List<Map<String, Object>> files);
        void onListFailed(String error);
    }

    public interface StorageCallback {
        void onStorageChecked(long[] storage);
        void onCheckFailed(String error);
    }

    public abstract void authenticate(int bucketNumber, String userId, AuthCallback callback);
    public abstract void listFiles(Integer maxResults, String query, ListFilesCallback callback);
    public abstract void checkStorage(StorageCallback callback);
}